spec/lib/gakubuchi/engine_registrar_spec.rb

```diff
@@ -1,71 +1,67 @@
+    let(:mime_type) { Gakubuchi::MimeType.new(content_type, extensions: extensions) }
+    let(:sprokcets_extensions) do
+      extension_type = major_version_of(Sprockets) >= 4 ? :transformers : :engines
+      -> { env.public_send(extension_type) }
+    end
 
     context "when specified engine is an uninitialized constant" do
-      let(:target) { :foo }
+      let(:content_type) { "application/csv+ruby" }
       let(:engine) { "Foo" }
+      let(:extensions) { %w(.rcsv) }
 
-      describe "env.engines" do
-        subject { -> { env.engines } }
-        it { expect(&described_method).not_to change(&subject) }
+      it "should return false" do
+        expect(described_method.call).to eq false
       end
 
-      describe "return value" do
-        subject { described_method.call }
-        it { is_expected.to eq false }
+      it "shouldn't register the engine for the MIME type" do
+        expect(&described_method).not_to change(&sprokcets_extensions)
       end
     end
 
-    context "when specified target is already registered" do
-      let(:target) { :sass }
-      let(:engine) { "Sprockets::SassTemplate" }
+    context "when there is no extensions corresponding the MIME type" do
+      let(:content_type) { "application/csv+ruby" }
+      let(:engine) { Tilt::CSVTemplate }
+      let(:extensions) { nil }
 
-      describe "env.engines" do
-        subject { -> { env.engines } }
-        it { expect(&described_method).not_to change(&subject) }
+      it "should return false" do
+        expect(described_method.call).to eq false
       end
 
-      describe "return value" do
-        subject { described_method.call }
-        it { is_expected.to eq false }
+      it "shouldn't register the engine for the MIME type" do
+        expect(&described_method).not_to change(&sprokcets_extensions)
       end
     end
 
-    context "when specified target is not registered" do
-      let(:target) { :foo }
-      let(:engine) { "Sprockets::SassTemplate" }
+    context "when all parameters are valid" do
+      let(:content_type) { "application/csv+ruby" }
+      let(:engine) { Tilt::CSVTemplate }
+      let(:extensions) {  %w(.rcsv) }
 
-      describe "env.engines" do
-        subject { -> { env.engines } }
-        let(:expectation) { a_hash_including(".foo" => Sprockets::SassTemplate) }
-
-        it { expect(&described_method).to change(&subject).to(expectation) }
+      it "should return true" do
+        expect(described_method.call).to eq true
       end
 
-      describe "return value" do
-        subject { described_method.call }
-        it { is_expected.to eq true }
-      end
-    end
-  end
-
-  describe "#registered?" do
-    subject { engine_registrar.registered?(target) }
+      it "should register the engine for the MIME type" do
+        diff =
+          case major_version_of(Sprockets)
+          when 2
+            { extensions.first => engine }
```

Comment:
[IMO]
You should use better way to specify the extension filter logic on [this  line](https://github.com/yasaichi/gakubuchi/pull/13/files#diff-c06ae655ad3364cd5cae9f4bfcd00cf1R39).